### PR TITLE
ensure encoding form dispose + store encoding in tag

### PR DESCRIPTION
**Pull request by ivandrofly**

### CODE REVIEW

File: src/Forms/BatchConvert.cs [hunk 1]
```diff
@@ -2681,20 +2681,22 @@ private void buttonBrowseEncoding_Click(object sender, EventArgs e)
             openFileDialog1.Filter = UiUtil.SubtitleExtensionFilter.Value;
             if (openFileDialog1.ShowDialog(this) == DialogResult.OK)
             {
-                var chooseEncoding = new ChooseEncoding();
-                chooseEncoding.Initialize(openFileDialog1.FileName);
-                if (chooseEncoding.ShowDialog(this) == DialogResult.OK)
+                using (var chooseEncoding = new ChooseEncoding())
```
Comment (ivandrofly) [inline src/Forms/BatchConvert.cs:2684]: add `using`

File: src/Forms/ChooseEncoding.cs [hunk 2]
```diff
@@ -63,11 +63,12 @@ public void Initialize(string fileName)
             foreach (var enc in Configuration.AvailableEncodings)
             {
                 var item = new ListViewItem(new[] { enc.CodePage.ToString(), enc.WebName, enc.EncodingName });
-                listView1.Items.Add(item);
+                item.Tag = enc;
```
Comment (ivandrofly) [inline src/Forms/ChooseEncoding.cs:66]: store encoding in item tag for faster retrieve later on...

File: src/Forms/Main.cs [hunk 3]
```diff
@@ -13102,13 +13102,15 @@ private void ToolStripMenuItemManualAnsiClick(object sender, EventArgs e)
             openFileDialog1.Filter = UiUtil.SubtitleExtensionFilter.Value;
             if (openFileDialog1.ShowDialog(this) == DialogResult.OK)
             {
-                var chooseEncoding = new ChooseEncoding();
-                chooseEncoding.Initialize(openFileDialog1.FileName);
-                if (chooseEncoding.ShowDialog(this) == DialogResult.OK)
+                using (var chooseEncoding = new ChooseEncoding())
```
Comment (ivandrofly) [inline src/Forms/Main.cs:13105]: add `using`

File: src/Forms/ChooseEncoding.cs (hunk 2 above)
Comment (niksedk) [reply src/Forms/ChooseEncoding.cs:66]: Is actually slower in my tests...

Comment (ivandrofly) [reply src/Forms/ChooseEncoding.cs:66]: That is strange... Encoding.GetEncoding is actually calling `EncodingProvider.GetEncodingFromProvider(name)` which loops through a array of encoding... maybe there is some cache of something I will investigate more!

image

image

**Pull request closed without merging**